schematic_api/api/security_controller_.py

```diff
@@ -1,14 +1,48 @@
-from typing import List
+import logging
+from typing import Dict, Union
 
+from jwt import PyJWKClient, decode
+from jwt.exceptions import InvalidTokenError
+from synapseclient import Synapse
 
-def info_from_bearerAuth(token):
+from schematic.configuration.configuration import CONFIG
+
+logger = logging.getLogger(__name__)
+
+syn = Synapse(
+    configPath=CONFIG.synapse_configuration_path,
+    cache_client=False,
+)
+jwks_client = PyJWKClient(
+    uri=syn.authEndpoint + "/oauth2/jwks", headers=syn._generate_headers()
+)
+
+
+def info_from_bearerAuth(token: str) -> Dict[str, Union[str, int]]:
     """
-    Check and retrieve authentication information from custom bearer token.
-    Returned value will be passed in 'token_info' parameter of your operation function, if there is one.
-    'sub' or 'uid' will be set in 'user' parameter of your operation function, if there is one.
-    :param token Token provided by Authorization header
-    :type token: str
-    :return: Decoded token information or None if token is invalid
-    :rtype: dict | None
+    Authenticate user using bearer token. The token claims are decoded and returned.
+
+    Example from:
+    <https://pyjwt.readthedocs.io/en/stable/usage.html#retrieve-rsa-signing-keys-from-a-jwks-endpoint>
+
+    Args:
+        token (str): Bearer token.
+
+    Returns:
+        dict: Decoded token information.
     """
-    return {"uid": "user_id"}
+    try:
+        jwks_client.get_signing_key_from_jwt(token)
+        signing_key = jwks_client.get_signing_key_from_jwt(token)
```

Comment:
Why are both `get_signing_key_from_jwt` calls needed?

Reply:
That was a mistake. I remove if